rflx/specification/parser.py

```diff
@@ -1565,12 +1572,32 @@ def create_model(self) -> model.Model:
             self.__evaluate_specification(error, spec_node.spec, spec_node.filename)
         try:
             result = model.Model(self.__types, self.__sessions)
+            self.__integration.validate(result, error)
         except RecordFluxError as e:
             error.extend(e)
 
         error.propagate()
         return result
 
+    def get_integration(self) -> Integration:
+        return self.__integration
+
+    def _load_integration_file(self, integration_file: Path, error: RecordFluxError) -> None:
+        if integration_file.exists():
+            yaml = YAML()
+            try:
+                content = yaml.load(integration_file)
+            except MarkedYAMLError as e:
+                location = None
+                if e.context_mark:
+                    location = Location(
+                        start=(e.context_mark.line, e.context_mark.column),
+                        source=e.context_mark.name,
+                    )
+                error.extend([(str(e), Subsystem.PARSER, Severity.ERROR, location)])
```

Comment:
For all my test cases, there is either `test.rfi:0:0` or no location:

```console
$ rflx check test.rflx
Parsing test.rflx
Parsing universal.rflx
Processing Universal
Processing Test
parser: error: mapping values are not allowed here
  in "test.rfi", line 3, column 20
```

```console
$ rflx check test.rflx
Parsing test.rflx
Parsing universal.rflx
Processing Universal
Processing Test
test.rfi:0:0: parser: error: while scanning a quoted scalar
  in "test.rfi", line 1, column 1
found unexpected end of stream
  in "test.rfi", line 2, column 1
```

Are there cases which work for you? I think there should be no case with no location, as we always know at least the affected file.

Reply:
Indeed the context_mark line/column seems to be always zero. By the way, is there any API documentation for ruamel.yaml? I looked at the source code ...
Also agreed that we always know the file at least.

Do you know how to extract the info from the ruamel exception? If not, let's just create a location with the file only (and a 0:0 or 1:1 location).

Reply:
ruamel.yaml is a fork of pyyaml, so the pyyaml documentation could help: https://pyyaml.org/wiki/PyYAMLDocumentation. It seems using `problem_mark` instead of `context_mark` could solve the problem.

Reply:
Could you please also extend the unit tests for cases where the location is not 0:0?

Reply:
you mean changing the regex, right? fixed in 674d7a7b864cee4b98d673fd1278a289458c50e0

Reply:
I meant not just matching any number in the regex, but checking that the expected line and column number is shown. Such a test would have detected that the version using `context_mark` did not work, and it would also detect if the current version stops working (e.g., because of changes in ruamel.yaml).

Reply:
done